packages/react-native-worklets/src/threads.ts

```diff
@@ -185,14 +192,27 @@ if (__DEV__ && !SHOULD_BE_USE_WEB) {
  * @returns The return value of the function passed as the first argument.
  * @see https://docs.swmansion.com/react-native-worklets/docs/threading/runOnUISync
  */
+// @ts-expect-error This overload is correct since it's what user sees in his code
+// before it's transformed by Worklets Babel plugin.
 export function runOnUISync<Args extends unknown[], ReturnValue>(
   worklet: (...args: Args) => ReturnValue,
   ...args: Args
+): ReturnValue;
+export function runOnUISync<Args extends unknown[], ReturnValue>(
+  worklet: WorkletFunction<Args, ReturnValue>,
+  ...args: Args
 ): ReturnValue {
-  return executeOnUIRuntimeSync(worklet)(...args);
+  return WorkletsModule.executeOnUIRuntimeSync(
+    createSerializable(() => {
+      'worklet';
+      const result = worklet(...args);
+      return makeShareableCloneOnUIRecursive(result);
+    })
+  );
 }
 
-// @ts-expect-error Check `executeOnUIRuntimeSync` overload above.
+// @ts-expect-error This overload is correct since it's what user sees in his code
```

Comment:
```suggestion
// @ts-expect-error This overload is correct since it's what user sees in their code
```

It is safer to use `their` these days 😄